src/audio/copier/copier_ipcgtw.c

```diff
@@ -250,3 +223,94 @@ void ipcgtw_zephyr_reset(struct comp_dev *dev)
 		comp_warn(dev, "ipcgtw_reset(): no buffer found");
 	}
 }
+
+int copier_ipcgtw_create(struct comp_dev *parent_dev, struct copier_data *cd,
+			 struct comp_ipc_config *config,
+			 const struct ipc4_copier_module_cfg *copier)
+{
+	struct ipcgtw_data *ipcgtw_data;
+	const struct ipc4_copier_gateway_cfg *gtw_cfg;
+	const struct ipc4_ipc_gateway_config_blob *blob;
+	int ret;
+
+	cd->ipc_gtw = true;
+
+	/* create_endpoint_buffer() uses this value to choose between input and
+	 * output formats in copier config to setup buffer. For this purpose
+	 * IPC gateway should be handled similarly as host gateway.
+	 */
+	config->type = SOF_COMP_HOST;
+
+	ret = create_endpoint_buffer(parent_dev, cd, config, copier, ipc4_gtw_none, false, 0);
+	if (ret < 0)
+		return ret;
+
+	gtw_cfg = &copier->gtw_cfg;
+	if (!gtw_cfg->config_length) {
+		comp_err(parent_dev, "ipcgtw_create(): empty ipc4_gateway_config_data");
+		ret = -EINVAL;
+		goto e_buf;
+	}
```

Comment:
can we check this before creating the buffer?

Reply:
ok, this is origin logic, I did not change it but if ahead of allocate endpoint buffer, we may need change error handing part.

Reply:
I'd also move lines 236 and 242 below this check. Why modify any external data if we conclude, that the configuration is invalid and error out?